extensions/wikia/Captcha/Captcha.class.php

```diff
@@ -0,0 +1,625 @@
+<?php
+
+namespace Captcha;
+
+use Captcha\Factory;
+use Wikia\Logger\WikiaLogger;
+
+class Handler {
+
+	// Number of seconds after a bad login that a captcha will be shown to that client on the login
+	// form to slow down password-guessing bots. Set to five minutes.
+	const BAD_LOGIN_TTL = 300;
+
+	// Number of bad login attempts before triggering the captcha.  0 means the captcha is presented
+	// on the first login.
+	const MAX_BAD_LOGIN_ATTEMPTS = 3;
+
+	protected $action;
+
+	protected $trigger;
+
+	/** @var \Captcha\Module\BaseCaptcha */
+	protected $captcha;
+
+	public function __construct() {
+		$this->captcha = Factory\Module::getInstance();
+	}
+
+	/**
+	 * Inject whazawhoo
+	 *
+	 * @fixme if multiple thingies insert a header, could break
```

Comment:
Let's create a ticket to have another pass at this extension for more cleanup like this. That way we can reference the ticket number in this comment, even if we don't get to it right away.